command/server/init.go

```diff
@@ -308,3 +315,56 @@ func (p *serverParams) initGRPCAddress() error {
 
 	return nil
 }
+
+// initBootnodeConfig initializes the bootnode configuration from various sources
+func (p *serverParams) initBootnodeConfig() error {
+	var err error
+	p.bootnodeConfig, err = p.getBootnodeConfig()
+	return err
+}
+
+// getBootnodeConfig retrieves bootnode configuration from various sources
+func (p *serverParams) getBootnodeConfig() (*chain.Bootnode, error) {
+	var bootnodes []string
+
+	// First try to load from specified bootnode path
+	if p.rawConfig.BootnodePath != "" {
+		// Load the bootnode configuration file directly
+		data, err := os.ReadFile(p.rawConfig.BootnodePath)
+		if err == nil {
+			var bootnodeConfig struct {
+				Bootnodes []string `json:"bootnodes"`
+			}
+			if err := json.Unmarshal(data, &bootnodeConfig); err == nil {
+				bootnodes = bootnodeConfig.Bootnodes
+			}
+		}
+	}
+
+	// If no bootnodes found and we're using mainnet/testnet, use their default bootnodes
+	if len(bootnodes) == 0 && (p.rawConfig.GenesisFile == "mainnet" || p.rawConfig.GenesisFile == "testnet") {
```

Comment:
Code improvement suggestion: you can leave only `len(bootnodes) == 0` in the first if and make an inner if to check for the mainnet/testnet case so the other `else if len(bootnodes) == 0`